src/ThreadsCommon.c

```diff
@@ -0,0 +1,164 @@
+#define RT_MAX_CYCLIC_INTERFACES RUNTIME_CYCLIC_INTERFACE_COUNT
+#else
+#define RT_MAX_CYCLIC_INTERFACES RUNTIME_THREAD_COUNT
+#endif
+
+#define EMPTY_REQUEST_DATA_BUFFER_SIZE 8
+
+#define NANOSECOND_IN_MILISECOND 1000000
+
+extern rtems_name generate_new_partition_timer_name();
+static void schedule_next_tick(const uint32_t cyclic_request_data_index);
+static void timer_callback(rtems_id timer_id, void *cyclic_request_data_index);
+static void update_execution_time_data(const uint32_t thread_id,
+				       const uint64_t thread_execution_time);
+
+typedef void (*call_function)(const char *buf, size_t len);
+
+struct CyclicRequestData {
+	rtems_id timer_id;
+	rtems_interval next_wakeup_ticks;
+	rtems_interval interval_ticks;
+	uint32_t queue_id;
+	uint32_t request_size;
+	bool is_used;
+};
+
+struct CyclicEmptyRequestData {
+	uint32_t m_sender_pid;
+	uint32_t m_length;
+	uint8_t m_data[EMPTY_REQUEST_DATA_BUFFER_SIZE]
+	    __attribute__((aligned(16)));
+};
+
+static struct CyclicRequestData cyclic_reqeust_data[RT_MAX_CYCLIC_INTERFACES];
+static struct CyclicEmptyRequestData empty_request;
+
+static void schedule_next_tick(const uint32_t cyclic_request_data_index)
+{
+	const rtems_id timer_id =
+	    cyclic_reqeust_data[cyclic_request_data_index].timer_id;
+
+	cyclic_reqeust_data[cyclic_request_data_index].next_wakeup_ticks +=
+	    cyclic_reqeust_data[cyclic_request_data_index].interval_ticks;
+	rtems_interval delta =
+	    cyclic_reqeust_data[cyclic_request_data_index].next_wakeup_ticks -
+	    rtems_clock_get_ticks_since_boot();
+	rtems_timer_fire_after(timer_id, delta, timer_callback,
+			       (void *)cyclic_request_data_index);
+}
+
+static void timer_callback(rtems_id timer_id, void *cyclic_request_data_index)
+{
+	uintptr_t index = (uintptr_t)cyclic_request_data_index;
+
+	rtems_message_queue_send((rtems_id)cyclic_reqeust_data[index].queue_id,
+				 &empty_request,
+				 cyclic_reqeust_data[index].request_size);
+	schedule_next_tick(index);
+}
+
+static void update_execution_time_data(const uint32_t thread_id,
+				       const uint64_t thread_execution_time)
+{
+
+	if (thread_execution_time <
+	    threads_info[thread_id].min_thread_execution_time) {
+		threads_info[thread_id].min_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	if (thread_execution_time >
+	    threads_info[thread_id].max_thread_execution_time) {
+		threads_info[thread_id].max_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	threads_info[thread_id].mean_thread_execution_time =
+	    threads_info[thread_id].mean_thread_execution_time +
+	    ((double)thread_execution_time -
+	     threads_info[thread_id].mean_thread_execution_time) /
+		((double)threads_info[thread_id].execution_time_counter + 1);
+
+	threads_info[thread_id].execution_time_counter++;
+}
+
+bool ThreadsCommon_CreateCyclicRequest(uint64_t interval_ns,
+				       uint64_t dispatch_offset_ns,
+				       uint32_t queue_id, uint32_t request_size)
+{
+	assert(request_size == 0);
```

Comment:
This assert will most probably fail, as the request size for cyclic interfaces should be 1....  I think the assert should be request_size <= EMPTY_REQUEST_DATA_BUFFER_SIZE

Reply:
Done